datapoint-forward/src/DataPoint/Forward/Utils.hs

```diff
@@ -0,0 +1,131 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+module DataPoint.Forward.Utils
+  ( DataPoint (..)
+  , DataPointStore
+  , DataPointAsker (..)
+  , initDataPointStore
+  , initDataPointAsker
+  , writeToStore
+  , readFromStore
+  , askForDataPoints
+  , runActionInLoop
+  ) where
+
+import           Control.Concurrent.STM (atomically, check)
+import           Control.Concurrent.STM.TVar
+import           Control.Exception (SomeAsyncException (..), fromException, tryJust)
+import           Control.Tracer (showTracing, stdoutTracer, traceWith)
+import           Data.Aeson
+import qualified Data.HashMap.Strict as HM
+import           System.Time.Extra (sleep)
+
+import           DataPoint.Forward.Configuration
+import           DataPoint.Forward.Protocol.Forwarder
+import           DataPoint.Forward.Protocol.Type
+
+-- | Type wrapper for some value of type 'v'. The only reason we need this
+--   wrapper is an ability to store different values in the same 'DataPointStore'.
```

Comment:
Note that this has consequences: when you read if from the store you will only be able to access it as unstructured `JSON` unlike Haskell algebraic data types.  It's probably fine for your application, but it might be worth mentioning the trade-off.